internal/zap/strings.go

```diff
@@ -12,3 +15,19 @@ func (ss Strings) MarshalLogArray(enc zapcore.ArrayEncoder) error {
 	}
 	return nil
 }
+
+// MapStringInterface is a map that implements MarshalLogObject.
+type MapStringInterface map[string]interface{}
+
+// MarshalLogObject implementation
+func (msi MapStringInterface) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	for key, val := range msi {
+		v, err := json.Marshal(val)
```

Comment:
Why do we need json.Marshal here?

Reply:
Because `val` is type `interface{}`, we cannot use it in `enc.AddString()`

Reply:
why not casting to `string`?

Reply:
Because It can be a nested object or an array.